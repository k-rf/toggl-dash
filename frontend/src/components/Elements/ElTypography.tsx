import { Typography, TypographyProps } from "@mui/material";

type Props = Pick<TypographyProps, "children" | "variant" | "color" | "sx">;

export const ElTypography = (props: Props) => {
  return <Typography {...props} />;
};
