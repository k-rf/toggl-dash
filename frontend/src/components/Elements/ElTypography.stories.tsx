import { Stack } from "@mui/material";
import type { Meta, StoryObj } from "@storybook/react";

import { ElTypography } from "./ElTypography";

export default { component: ElTypography } as Meta<typeof ElTypography>;

type Story = StoryObj<typeof ElTypography>;

export const Default: Story = {
  render: () => {
    return (
      <Stack spacing={2}>
        <ElTypography variant="h1">H1</ElTypography>
        <ElTypography variant="h2">H2</ElTypography>
        <ElTypography variant="h3">H3</ElTypography>
        <ElTypography variant="h4">H4</ElTypography>
        <ElTypography variant="h5">H5</ElTypography>
        <ElTypography variant="h6">H6</ElTypography>
        <ElTypography variant="body1">Body1</ElTypography>
        <ElTypography variant="body2">Body2</ElTypography>
        <ElTypography variant="caption">Caption</ElTypography>
        <ElTypography variant="subtitle1">Subtitle1</ElTypography>
        <ElTypography variant="subtitle2">Subtitle2</ElTypography>
        <ElTypography variant="overline">Overline</ElTypography>
      </Stack>
    );
  },
};
