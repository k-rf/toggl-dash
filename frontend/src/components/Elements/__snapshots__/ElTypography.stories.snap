// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Elements/ElTypography > Default 1`] = `
<div
  className="css-1p5q5e5-MuiStack-root"
>
  <h1
    className="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
  >
    H1
  </h1>
  <h2
    className="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
  >
    H2
  </h2>
  <h3
    className="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
  >
    H3
  </h3>
  <h4
    className="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
  >
    H4
  </h4>
  <h5
    className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
  >
    H5
  </h5>
  <h6
    className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
  >
    H6
  </h6>
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    Body1
  </p>
  <p
    className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
  >
    Body2
  </p>
  <span
    className="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
  >
    Caption
  </span>
  <h6
    className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
  >
    Subtitle1
  </h6>
  <h6
    className="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
  >
    Subtitle2
  </h6>
  <span
    className="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
  >
    Overline
  </span>
</div>
`;
